src/automata.jl

```diff
@@ -307,14 +335,13 @@ to `v`. For standard rewriting `v` should be empty.
 """
 function rewrite_from_left!(v::AbstractWord, w::AbstractWord, a::Automaton)
     past_states = a._past_states
-    resize!(past_states, length(w))
+    resize!(past_states, length(w) + 1)
     state = initialstate(a)
     past_states[1] = state
-    initial_length = length(v)
 
     while !isone(w)
         x = popfirst!(w)
-        k = length(v) - initial_length + 1
+        k = length(v) + 1
         state = outedges(past_states[k])[x]
```

Comment:
why this change? If `v` is non-empty (we're just appending the rewrite procedure to existing `v`) this will fail:
`length(w)` might be small compared to `length(v)` and we get out of bounds

Reply:
Ahh, yes that's true. I forgot for a moment about the way interface to rewriting is designed. `k = length(v) - initial_length + 1` will be ok. Sorry for that.